pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -85,34 +85,11 @@ func writeVifFile(buf []byte, pid, name string) error {
 }
 
 func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid string) error {
-	link, err := Handler.LinkByName(podInterfaceName)
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
-		return err
-	}
-	// get IP address
-	addrList, err := Handler.AddrList(link, netlink.FAMILY_ALL)
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to get a address for interface: %s", podInterfaceName)
-		return err
-	}
-	// no ip assigned. ipam disabled
-	if len(addrList) == 0 {
-		return nil
-	}
-
 	cache := PodCacheInterface{Iface: iface}
 
-	ipv4 := ""
-	ipv6 := ""
-	for _, addr := range addrList {
-		if addr.IP.IsGlobalUnicast() {
-			if netutils.IsIPv6(addr.IP) {
-				ipv6 = addr.IP.String()
-			} else {
-				ipv4 = addr.IP.String()
-			}
-		}
+	ipv4, ipv6, err := readIPAddressesFromLink(podInterfaceName)
```

Comment:
Actually it is better to leave all the code inside `readIPAddressesFromLink` and export the podIPS ordering code to another function
And  please add comment about why it is necessary to sort podIPS, we want to make dev life easy by not make them dig in git blame until they find the right commit message.

Reply:
We talked couple people, several times about comments.
Edy agreed with me about that, i wont add it unless other agree with you, please we shouldn't raise again and again a subject in differ threads imo, at least when it was talked there.

I am working on the switch that Edy asked, and then we will see what the reviewers decide together.

Reply:
> You mean leave readIPAddressesFromLink as is and export the suggested "switch" code to its own function as well?

Exactly

> You can suggest a function name that self explain this without the comments, like "sortBasedOnPrimaryIPVersion" or something like that. If someone is interested in the full story, he can go to the blame.
> Will that be good enough?

It is better, but it is not explaining why we need to sort is.

Please rephrase the commit message the part that explains why this is needed and add reference (if there is..), sorry but its still not simple to understand it from the commit.
And add a comment in the code that explain why this is needed (sort podIPS) with few words

Reply:
> You mean leave readIPAddressesFromLink as is and export the suggested "switch" code to its own function as well?


Exactly

> You can suggest a function name that self explain this without the comments, like "sortBasedOnPrimaryIPVersion" or something like that. If someone is interested in the full story, he can go to the blame.
> Will that be good enough?

It is better, but it is not explaining why we need to sort is.

If you want you can leave it in one function.
Please add comment in the code that explain why this is needed (sort podIPS) with few words

Reply:
Funny thing is that we just duplicate kubelet logic here and it would have been avoided by copying the list from the pod IP list as is for the primary/pod network.

"sortIPBasedOnPodSortingLogic" ?

Maybe you should suggest the code comment you expect and we can then figure out what to do.

Reply:
Done (all)
testing it locally meanwhile